packages/design-system-react-native/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+import type { ButtonProps } from './Button.types';
+import { ButtonVariant } from './Button.types';
+import ButtonPrimary from './variants/ButtonPrimary';
+import ButtonSecondary from './variants/ButtonSecondary';
+import ButtonTertiary from './variants/ButtonTertiary';
+
+const Button = (buttonProps: ButtonProps) => {
+  const { variant, ...restProps } = buttonProps;
+
+  switch (variant) {
+    case ButtonVariant.Tertiary:
+      return <ButtonTertiary {...restProps} />;
+    case ButtonVariant.Primary:
+      return <ButtonPrimary {...restProps} />;
+    case ButtonVariant.Secondary:
+      return <ButtonSecondary {...restProps} />;
+    default:
+      throw new Error('Invalid Button Variant');
```

Comment:
non-blocking suggestion: consider adding a more description error message 
```suggestion
      throw new Error(`Invalid Button Variant: ${variant}. Expected one of: ${Object.values(ButtonVariant).join(', ')}`);
```

<img width="300" alt="Screenshot 2025-02-06 at 9 16 48 AM" src="https://github.com/user-attachments/assets/0d2fe751-6fb2-46a4-b014-2418a5f5e521" />